src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -170,21 +171,50 @@ private JvmObjectReference CreateBroadcast(SparkContext sc, T value)
             bool encryptionEnabled = bool.Parse(
                 sc.GetConf().Get("spark.io.encryption.enabled", "false"));
 
+            var _pythonBroadcast = (JvmObjectReference)javaSparkContext.Jvm.CallStaticJavaMethod(
+                "org.apache.spark.api.python.PythonRDD",
+                "setupBroadcast",
+                _path);
+
             if (encryptionEnabled)
             {
-                throw new NotImplementedException("Broadcast encryption is not supported yet.");
+                var pair = (JvmObjectReference[])_pythonBroadcast.Invoke("setupEncryptionServer");
+
+                using ISocketWrapper socket = SocketFactory.CreateSocket();
+                socket.Connect(
+                    IPAddress.Loopback,
+                    (int)pair[0].Invoke("intValue"),
+                    (string)pair[1].Invoke("toString"));
+                WriteEncrypted(value, socket.OutputStream);
+                socket.OutputStream.Close();
+                _pythonBroadcast.Invoke("waitTillDataReceived");
```

Comment:
Does the following work?
```csharp
using (ISocketWrapper socket = SocketFactory.CreateSocket())
{
    socket.Connect(
        IPAddress.Loopback,
        (int)pair[0].Invoke("intValue"),
        (string)pair[1].Invoke("toString"));
    WriteEncrypted(value, socket.OutputStream);
}
_pythonBroadcast.Invoke("waitTillDataReceived");
```
I am trying to understand of the scope of `socket` and `socket.OutputStream`. Also would `Flush` work in place of `Close`?

Reply:
Changed as requested, works fine. `Flush` works in place of `Close`